engines/hypno/scene.cpp

```diff
@@ -0,0 +1,371 @@
+}
+
+// Hotspots
+
+void HypnoEngine::clickedHotspot(Common::Point mousePos) {
+	Hotspots *hots = stack.back();
+	Hotspot selected;
+	bool found = false;
+	int rs = 100000000;
+	int cs = 0;
+	for (Hotspots::const_iterator it = hots->begin(); it != hots->end(); ++it) {
+		const Hotspot h = *it;
+		if (h.type != MakeHotspot)
+			continue;
+
+		cs = h.rect.width() * h.rect.height();
+		if (h.rect.contains(mousePos)) {
+			if (cs < rs) {
+				selected = h;
+				found = true;
+				rs = cs;
+			}
+		}
+	}
+	if (found) {
+		if (selected.smenu) {
+			assert(!selected.smenu->empty());
+			_nextHotsToAdd = selected.smenu;
+		}
+
+		for (Actions::const_iterator itt = selected.actions.begin(); itt != selected.actions.end(); ++itt) {
+			Action *action = *itt;
+			if (typeid(*action) == typeid(ChangeLevel))
+				runChangeLevel((ChangeLevel *)action);
+			if (typeid(*action) == typeid(Escape))
+				runEscape((Escape *)action);
+			else if (typeid(*action) == typeid(Cutscene))
+				runCutscene((Cutscene *)action);
+			else if (typeid(*action) == typeid(Play))
+				runPlay((Play *)action);
+			else if (typeid(*action) == typeid(WalN))
+				runWalN((WalN *)action);
+			else if (typeid(*action) == typeid(Global))
+				runGlobal((Global *)action);
+			else if (typeid(*action) == typeid(Talk))
+				runTalk((Talk *)action);
+			else if (typeid(*action) == typeid(Quit))
+				runQuit((Quit *)action);
+			else if (typeid(*action) == typeid(Palette))
+				debugC(1, kHypnoDebugScene, "runPalette unimplemented");
+		}
+	}
+}
+
+bool HypnoEngine::hoverHotspot(Common::Point mousePos) {
+	Hotspots *hots = stack.back();
+	Hotspot selected;
+	bool found = false;
+	int rs = 100000000;
+	int cs = 0;
+	for (Hotspots::const_iterator it = hots->begin(); it != hots->end(); ++it) {
+		const Hotspot h = *it;
+		if (h.type != MakeHotspot)
+			continue;
+
+		cs = h.rect.width() * h.rect.height();
+		if (h.rect.contains(mousePos)) {
+			if (cs < rs) {
+				selected = h;
+				found = true;
+				rs = cs;
+			}
+		}
+	}
+	if (found) {
+		for (Actions::const_iterator itt = selected.actions.begin(); itt != selected.actions.end(); ++itt) {
+			Action *action = *itt;
+			if (typeid(*action) == typeid(Mice))
+				runMice((Mice *)action);
+		}
+		return true;
+	}
+	return false;
+}
+
+void HypnoEngine::runTransition(Transition trans) {
+	for (Filenames::iterator it = trans.intros.begin(); it != trans.intros.end(); ++it) {
+		MVideo v(*it, Common::Point(0, 0), false, true, false);
+		runIntro(v);
+	}
+
+	if (!trans.frameImage.empty()) {
+		debugC(1, kHypnoDebugScene, "Rendering %s frame in transaction", trans.frameImage.c_str());
+		Graphics::Surface *frame = decodeFrame(trans.frameImage, trans.frameNumber);
+		Graphics::Surface *sframe = frame->scale(_screenW, _screenH);
+		drawImage(*sframe, 0, 0, false);
+		drawScreen();
+		frame->free();
+		delete frame;
+		sframe->free();
+		delete sframe;
+		Common::String *ptr = new Common::String(trans.level);
+		assert(installTimer(2 * 1000000, ptr));
```

Comment:
`assert()` will be removed in RELEASE build on Windows, thus, it must have no side effects. In your case, in the release builds you will have no timer installed :)

Reply:
Fixed.